labgrid/driver/power/gude24.py

```diff
@@ -25,6 +25,8 @@ def power_set(host, port, index, value):
         "http://{}:{}/ov.html?cmd=1&p={}&s={}".format(host, port, index, value)
     )
     r.raise_for_status()
+    if "Device is blocked by another user" in r.text:
+        raise ExecutionError("Could not set power, device is blocked by other user.")
```

Comment:
I wonder if instead of grepping for a specific problem it would be wiser to (at least try) to grep for any error and use its text in the Exception to be raised. This also had the advantage that the IP address is included in the message.

Having said that, using a method that allows error handling in a more reliable way would be great. There is http://wiki.gude.info/EPC_HTTP_Interface which documents a json interface. This isn't however available on the network power ports I have access to. :-|

Reply:
A better approach would be to use PDUDaemon after https://github.com/pdudaemon/pdudaemon/pull/78 or something similar is merged. The requests are serialized and come from the same IP.
As a workaround, we have a socat forward on the exporter for our lab.

Reply:
I agree that just filtering for this one exception is not sufficient. I made some bigger changes to the driver, that allow to check if the set method changed the value correctly without code duplication. With this implementation the whole response will be printed if the state of the ports can not be checked.